File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/pat/create/PersonalAccessTokenCreation.java
package ch.ethz.sis.openbis.generic.asapi.v3.dto.pat.create;

import java.util.Date;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.ObjectToString;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.ICreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.create.IObjectCreation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.person.id.IPersonId;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author pkupczyk
 */
@JsonObject("as.dto.pat.create.PersonalAccessTokenCreation")
public class PersonalAccessTokenCreation implements ICreation, IObjectCreation
{
    private static final long serialVersionUID = 1L;

    private IPersonId ownerId;

    private String sessionName;

    private Date validFromDate;

    private Date validToDate;

    public String getSessionName()
    {
        return sessionName;
    }

    public void setSessionName(final String sessionName)
    {
        this.sessionName = sessionName;
    }

    public Date getValidFromDate()
    {
        return validFromDate;
    }

    public void setValidFromDate(final Date validFromDate)
    {
        this.validFromDate = validFromDate;
    }

    public Date getValidToDate()
    {
        return validToDate;
    }

    public void setValidToDate(final Date validToDate)
    {
        this.validToDate = validToDate;
    }

    public IPersonId getOwnerId()
    {
        return ownerId;
    }

    public void setOwnerId(final IPersonId ownerId)
    {
        this.ownerId = ownerId;
    }

    @Override
    public String toString()
    {
        return new ObjectToString(this).append("ownerId", ownerId).append("sessionName", sessionName).append("validFrom", validFromDate)
                .append("validTo", validToDate).toString();
    }

}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/pat/create/PersonalAccessTokenCreation.js
/**
 * @author pkupczyk
 */
define([ "stjs" ], function(stjs) {
	var PersonalAccessTokenCreation = function() {
	};
	stjs.extend(PersonalAccessTokenCreation, null, [], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.pat.create.PersonalAccessTokenCreation';
		constructor.serialVersionUID = 1;
		prototype.ownerId = null;
		prototype.sessionName = null;
		prototype.validFromDate = null;
		prototype.validToDate = null;

		prototype.getOwnerId = function() {
			return this.ownerId;
		};
		prototype.setOwnerId = function(ownerId) {
			this.ownerId = ownerId;
		};
		prototype.getSessionName = function() {
			return this.sessionName;
		};
		prototype.setSessionName = function(sessionName) {
			this.sessionName = sessionName;
		};
		prototype.getValidFromDate = function() {
			return this.validFromDate;
		};
		prototype.setValidFromDate = function(validFromDate) {
			this.validFromDate = validFromDate;
		};
		prototype.getValidToDate = function() {
			return this.validToDate;
		};
		prototype.setValidToDate = function(validToDate) {
			this.validToDate = validToDate;
		};
	}, {
		ownerId : "IPersonId",
		validFromDate : "Date",
		validToDate : "Date"
	});
	return PersonalAccessTokenCreation;
})